mesonbuild/environment.py

```diff
@@ -1179,3 +1154,53 @@ def miss_defaulting(self):
 
     def detect_build(self, compilers = None):
         self.build = MachineInfo.detect(compilers)
+
+class BinaryTable:
+    # Map from language identifiers to environment variables.
+    evarMap = {
+        # Compilers
+        'c': 'CC',
+        'cpp': 'CXX',
+        'cs': 'CSC',
+        'd': 'DC',
+        'fortran': 'FC',
+        'objc': 'OBJC',
+        'objcpp': 'OBJCXX',
+        'rust': 'RUSTC',
+        'vala': 'VALAC',
+
+        # Binutils
+        'strip': 'STRIP',
+        'ar': 'AR',
+    }
```

Comment:
I figure this is the first step to centralizing the snooping of environment variables, which is a good thing so that impurity is easier to audit.